src/share/poudriere/common.sh

```diff
@@ -6646,6 +6689,37 @@ listed_ports() {
 	fi
 	_listed_ports "$@"
 }
+
+_list_ports_dir() {
+	local ptdir=$1
+	# skip overlays with no categories listed
+	[ -f ${ptdir}/Makefile ] || return
+	for cat in $(awk -F= '$1 ~ /^[[:space:]]*SUBDIR[[:space:]]*\+/ {gsub(/[[:space:]]/, "", $2); print $2}' ${ptdir}/Makefile); do
+		# skip overlays with no ports hooked to the build
+		[ -f ${ptdir}/${cat}/Makefile ] || continue
+		awk -F= -v cat=${cat} '$1 ~ /^[[:space:]]*SUBDIR[[:space:]]*\+/ {gsub(/[[:space:]]/, "", $2); print cat"/"$2}' ${ptdir}/${cat}/Makefile
+	done | while mapfile_read_loop_redir origin; do
+		if ! [ -d "${ptdir}/${origin}" ]; then
+			msg_warn "Nonexistent origin listed in category Makefiles: ${COLOR_PORT}${origin}${COLOR_RESET} (skipping)"
```

Comment:
Anywhere there's a warning or assertion for a problem in a port now needs to display which `OVERLAY` it's in.

Reply:
done